### ISSUE TITLE
Remove WebcomponentsJs lib

### ISSUE THREAD

**Issue by alansemenov** [enonic/lib-admin-ui#998]
We'll migrate Context panel from HTML Imports to Fetch API, so this lib is no longer relevant.

**Comment by sigdestad** [comment]
I have serious problems understanding how fetch API and webComponents are at all related?
How do we plan to isolate html and javascript in "widgets" from the rest of the page?

**Comment by alansemenov** [comment]
We are using HTML Imports (which is one of four Webcomponent standards) to fetch HTML content of widgets and embed it into the widget panel (the same is used to display the Launcher panel). Since HTML Imports are deprecated and will soon be removed, we need to migrate, and using fetch API is the easiest migration path. It doesn't have anything to do with encapsulation of CSS or JS, but we've never had it anyway.

If we want encapsulation, we need to research implementing this natively, without using JS frameworks, the best candidate is Shadow DOM v1.

**Comment by sigdestad** [comment]
Yes, we should minimum have a look at this at the same time - currently, there is nothing that prevents widgets from breaking our entire content studio app when they are loaded.

**Comment by sigdestad** [comment]
ES modules can isolate javascript, and shadow dom HTML I assume?

**Comment by alansemenov** [comment]
We're discussing two separate issues now: 1) be able to fetch HTML contents of widget or launcher panel via URL, 2) make sure that when we embed these fetched contents they are isolated from the rest of the host page (encapsulation of JS and CSS).

The first issue is now solved with HTML Imports. They are deprecated and we have to migrate. Using Fetch API is a no-brainer here - it's natively supported, it's fast to fetch the data and it's fast to migrate to (took me about an hour to make a [PR for the launcher panel](https://github.com/enonic/app-admin-home/pull/64)). Other alternatives are ajax (no) or iframes (no).

The second issue is more complicated and has to be researched. And yes, we have to do this even though it has been working without encapsulation for 3 years now. Seems like Shadow DOM is finally supported by most of the browsers except for Edge where it's under development.
